pkg/cas/resolver/resolver.go

```diff
@@ -108,6 +178,25 @@ func (h *Resolver) getAndDataFromRemote(webCASEndpoint *url.URL, cid string) ([]
 	return responseBody, nil
 }
 
+func (h *Resolver) getAndStoreDataFromIPFS(cid string) ([]byte, error) {
+	if h.ipfsReader == nil {
+		return nil, fmt.Errorf("ipfs reader is not supported")
```

Comment:
This error should actually be cas.ErrContentNotFound, because if you reach here it means that the content couldn't be found everywhere else, and IPFS not being enabled shouldn't be an error anyway since it's optional

Reply:
This is more configuration error so I prefer to leave it as is. We should not be getting "ipfs" hints and not support them.

Reply:
The problem is on line 98. You have a "last resort" type of check that, when failing, will result in a "not supported" msg

Reply:
Actually, that "not supported" msg won't happen, but at line 96 you check if h.ipfsReader isn't nil, and then in the method here you immediately check to see if it is nil, so I suggest moving the nil check outside the method

Reply:
I don't see any issue with doing two nil checks, however I will move the code as suggested.

Reply:
Done